lib/bunq/resource.rb

```diff
@@ -52,17 +54,34 @@ def append(path)
     end
 
     def ensure_session!
-      client.ensure_session!
+      client.current_session ||= client.session_servers.create
```

Comment:
`client.ensure_session!`  already caches the session in the `client`. So `ensure_session!` can simply propagate to the client`